crowbar_framework/lib/crowbar/deployment_queue.rb

```diff
@@ -374,10 +374,9 @@ def elements_not_ready(nodes, pre_cached_nodes = {})
       # Check to see if we should delay our commit until nodes are ready.
       delay = []
       nodes.each do |n|
-        node = Node.find_by_name(n)
+        pre_cached_nodes[n] ||= Node.find_by_name(n)
+        node = pre_cached_nodes[n]
```

Comment:
Since you're fixing things in this file: you may want to do a followup PR to use the cache in `elements_to_nodes_to_roles_map` too...

Reply:
done!